### PR TITLE
[Experiment] Support `godoc` by replacing the `pocket `module name with `github.com/pokt-network/poktroll`

### PR DESCRIPTION
**Pull request by Olshansk**
I started trying to explore how to get `godoc` documentation for our library and went down a rabbithole...

In short, in order for it to be available at https://pkg.go.dev/github.com/pokt-network/poktroll, similar to https://pkg.go.dev/github.com/pokt-network/pocket, we need to replace the `pocket` module name with `github.com/pokt-network/poktroll`, which affects everything.

Screenshot 2023-11-01 at 5 09 54 PM

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -48,16 +48,27 @@ go_version_check:
...
 		exit 1; \
 	fi; \
 	}
 
+.PHONY: check_godoc
+# Internal helper target - check if godoc is installed
+check_godoc:
+	{ \
+	if ( ! ( command -v godoc >/dev/null )); then \
+		echo "Seems like you don't have godocinstalled. Make sure you install it via 'go install golang.org/x/tools/cmd/godoc@latest' before continuing"; \
```
Comment (bryanchriswhite) [inline Makefile:66]:
```suggestion
		echo "Seems like you don't have godoc installed. Make sure you install it via 'go install golang.org/x/tools/cmd/godoc@latest' before continuing"; \
```

File: Makefile [hunk 2]
```diff
@@ -35,9 +35,9 @@ help: ## Prints all the targets in all the Makefiles
 ### Checks ###
 ##############
 
-.PHONY: go_version_check
+.PHONY: check_go_version
 # Internal helper target - check go version
-go_version_check:
+check_go_version:
```
Comment (bryanchriswhite) [inline Makefile:40]: :ok_hand:

File: docs/static/openapi.yml
Comment (bryanchriswhite) [inline docs/static/openapi.yml:1]: Why was this deleted?

File: internal/testclient/testeventsquery/connection.go [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
 import (
-	"pocket/pkg/either"
 	"testing"
 
 	"github.com/golang/mock/gomock"
 
-	"pocket/internal/mocks/mockclient"
+	"github.com/pokt-network/poktroll/internal/mocks/mockclient"
+	"github.com/pokt-network/poktroll/pkg/either"
```
Comment (bryanchriswhite) [inline internal/testclient/testeventsquery/connection.go:9]: :astonished: :pray:

File: pkg/observable/channel/observable.go [hunk 4]
```diff
@@ -2,7 +2,8 @@ package channel
 
 import (
 	"context"
-	"pocket/pkg/observable"
+
+	"github.com/pokt-network/poktroll/pkg/observable"
```
Comment (bryanchriswhite) [inline pkg/observable/channel/observable.go:6]: :pray:

Comment (bryanchriswhite) [inline pkg/observable/channel/observable.go:6]: :pray:

File: testutil/keeper/application.go [hunk 5]
```diff
@@ -15,10 +15,11 @@ import (
...
 
-	mocks "pocket/testutil/application/mocks"
-	"pocket/x/application/keeper"
-	"pocket/x/application/types"
-	gatewaytypes "pocket/x/gateway/types"
+	mocks "github.com/pokt-network/poktroll/testutil/application/mocks"
+
+	"github.com/pokt-network/poktroll/x/application/keeper"
+	"github.com/pokt-network/poktroll/x/application/types"
+	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
```
Comment (bryanchriswhite) [inline testutil/keeper/application.go:22]:
```suggestion
	mocks "github.com/pokt-network/poktroll/testutil/application/mocks"
	"github.com/pokt-network/poktroll/x/application/keeper"
	"github.com/pokt-network/poktroll/x/application/types"
	gatewaytypes "github.com/pokt-network/poktroll/x/gateway/types"
```

File: testutil/keeper/pocket.go [hunk 6]
```diff
@@ -12,9 +12,9 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	typesparams "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/pokt-network/poktroll/x/pocket/keeper"
+	"github.com/pokt-network/poktroll/x/pocket/types"
 	"github.com/stretchr/testify/require"
```
Comment (bryanchriswhite) [inline testutil/keeper/pocket.go:17]:
```suggestion
	storetypes "github.com/cosmos/cosmos-sdk/store/types"
	sdk "github.com/cosmos/cosmos-sdk/types"
	typesparams "github.com/cosmos/cosmos-sdk/x/params/types"
	"github.com/stretchr/testify/require"
	
	"github.com/pokt-network/poktroll/x/pocket/keeper"
	"github.com/pokt-network/poktroll/x/pocket/types"
```

File: testutil/keeper/service.go [hunk 7]
```diff
@@ -12,9 +12,9 @@ import (
 	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	typesparams "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/pokt-network/poktroll/x/service/keeper"
+	"github.com/pokt-network/poktroll/x/service/types"
 	"github.com/stretchr/testify/require"
```
Comment (bryanchriswhite) [inline testutil/keeper/service.go:17]:
```suggestion
	typesparams "github.com/cosmos/cosmos-sdk/x/params/types"
	"github.com/stretchr/testify/require"
	
	"github.com/pokt-network/poktroll/x/service/keeper"
	"github.com/pokt-network/poktroll/x/service/types"
```

File: docs/static/openapi.yml [hunk 8]
```diff
@@ -47201,189 +47201,13 @@ paths:
                             ApplicationServiceConfig holds the service
```
Comment (Olshansk) [inline docs/static/openapi.yml:47201]: @bryanchriswhite Note that this was updated, not deleted.

**Review by bryanchriswhite — APPROVED** [review]
:100::tada::raised_hands: tysm for doing this! :ship:
